src/main/java/io/jsonwebtoken/impl/DefaultJwtBuilder.java

```diff
@@ -252,7 +252,24 @@ public JwtBuilder claim(String name, Object value) {
 
         return this;
     }
+    
+    /**
+     * Add custom claims to the jwt payload
+     * @param customClaims map containing custom claims
+     * @return JwtBuilder
+     */
+    public JwtBuilder appendCustomClaims(Map<String, Object> customClaims){
+        if(customClaims.isEmpty()){
```

Comment:
This will throw an NPE if `customClaims` is null.  Instead, I'd do the following:

```java
Assert.notEmpty(customClaims, "Claims map cannot be null or empty.");
```

Additionally, this approach eliminates the if/else flow, which is easier to read and reduces cyclomatic complexity.  The implementation should look like this:

```java
Assert.notEmpty(customClaims, "Claims map cannot be null or empty.");

for (Map.Entry<String, Object> c : customClaims.entrySet()) {
    claim(c.getKey(), c.getValue());
}

return this;
```

Also, please change the method signature to be:

```java
public JwtBuilder addClaims(Map<String,Object> claims);
```

('add' is more aligned with Java Collections naming semantics.  It is also congruently named with the `setClaims` method)

Can you please update the PR accordingly?  Thank you!